File: uiza-player/src/main/java/io/uiza/player/widgets/previewseekbar/PreviewView.java
package io.uiza.player.widgets.previewseekbar;
import android.widget.FrameLayout;

import androidx.annotation.ColorInt;
import androidx.annotation.ColorRes;

public interface PreviewView {
    int getProgress();

    int getMax();

    int getThumbOffset();

    int getDefaultColor();

    boolean isShowingPreview();

    void showPreview();

    void hidePreview();

    void setPreviewLoader(PreviewLoader previewLoader);

    void setPreviewColorTint(@ColorInt int color);

    void setPreviewColorResourceTint(@ColorRes int color);

    void attachPreviewFrameLayout(FrameLayout frameLayout);

    void addOnPreviewChangeListener(OnPreviewChangeListener listener);

    void removeOnPreviewChangeListener(OnPreviewChangeListener listener);

    interface OnPreviewChangeListener {
        void onStartPreview(PreviewView previewView, int progress);

        void onStopPreview(PreviewView previewView, int progress);

        void onPreview(PreviewView previewView, int progress, boolean fromUser);
    }
}

File: uiza-player/src/main/java/io/uiza/player/widgets/previewseekbar/PreviewView.kt
package io.uiza.player.widgets.previewseekbar

import android.widget.FrameLayout
import androidx.annotation.ColorInt
import androidx.annotation.ColorRes

interface PreviewView {
    fun getProgress(): Int
    fun getMax(): Int
    fun getThumbOffset(): Int
    fun getDefaultColor(): Int
    fun isShowingPreview(): Boolean
    fun showPreview()
    fun hidePreview()
    fun setPreviewLoader(previewLoader: PreviewLoader?)
    fun setPreviewColorTint(@ColorInt color: Int)
    fun setPreviewColorResourceTint(@ColorRes color: Int)
    fun attachPreviewFrameLayout(frameLayout: FrameLayout?)
    fun addOnPreviewChangeListener(listener: OnPreviewChangeListener?)
    fun removeOnPreviewChangeListener(listener: OnPreviewChangeListener?)
    interface OnPreviewChangeListener {
        fun onStartPreview(previewView: PreviewView?, progress: Int)
        fun onStopPreview(previewView: PreviewView?, progress: Int)
        fun onPreview(
            previewView: PreviewView?,
            progress: Int,
            fromUser: Boolean
        )
    }
}